### PR TITLE
Make mission system work

### PR DESCRIPTION
**Pull request by GabrielFleischer**
Improve mission system such that it handles more edge cases and provide more feedback

### CODE REVIEW

**Review by Anghul — APPROVED** [review]
I didn't see anything off

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -97,9 +97,10 @@
 
     <!-- Drone -->
     <string name="drone_mission_success">The mission was successfully dispatched to the drone</string>
-    <string name="drone_mission_error">An error occurred while dispatching missing to the drone</string>
+    <string name="drone_mission_error">An error occurred while dispatching missing to the drone : %s</string>
     <string name="drone_pause_success">Mission paused</string>
     <string name="drone_pause_error">Mission pause failed, try again later</string>
```
Comment (jonasblanc) [inline app/src/main/res/values/strings.xml:102]: The resource `R.string.drone_pause_error` appears to be unused. You could remove it.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -97,9 +97,10 @@
...
     <string name="drone_return_error">Could not launch mission to let the drone come back</string>
     <string name="drone_return_success">The drone is coming back</string>
```
Comment (jonasblanc) [inline app/src/main/res/values/strings.xml:104]: The resource `R.string.drone_return_success` appears to be unused. You could remove it.

**Review by jonasblanc — CHANGES_REQUESTED** [review]

**Comment by jonasblanc** [comment]
Great code !

**Review by jonasblanc — APPROVED** [review]

**Pull request merged**